ntoskrnl/ke/thrdobj.c

```diff
@@ -1137,7 +1139,7 @@ KeSetSystemAffinityThread(IN KAFFINITY Affinity)
         }
 
         /* Calculate the ideal CPU from the affinity set */
-        BitScanReverse(&NodeMask, AffinitySet);
+        BitScanReverseAffinity(&IdealProcessor, AffinitySet);
         CurrentThread->IdealProcessor = (UCHAR)NodeMask;
```

Comment:
Obtained `IdealProcessor` value being unused?

Reply:
Oh, no that should be used instead of `NodeMask`.